src/Features/Core/Portable/SolutionCrawler/IdleProcessor.cs

```diff
@@ -85,10 +100,10 @@ private async Task ProcessAsync()
                     // wait for next item available
                     await WaitAsync(this.CancellationToken).ConfigureAwait(continueOnCapturedContext: false);
 
-                    using (this.Listener.BeginAsyncOperation("ProcessAsync"))
+                    using (var asyncToken = this.Listener.BeginAsyncOperation("ProcessAsync"))
                     {
                         // we have items but workspace is busy. wait for idle.
-                        await WaitForIdleAsync().ConfigureAwait(continueOnCapturedContext: false);
+                        await WaitForIdleAsync(asyncToken).ConfigureAwait(continueOnCapturedContext: false);
```

Comment:
Mentally it's very strange to me we're passing the token here and not the listener. It implies that somehow to me the wait is dependent on the token which isn't the case at all.

Reply:
➡️ Now passing the listener as an `IExpeditableDelaySource`